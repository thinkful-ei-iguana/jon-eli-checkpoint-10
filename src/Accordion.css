body{
  background-color: black;
}

ul {
  list-style-type: armenian;
}


li {
  color: firebrick;
  
}

li + li {
  margin-top: 27px;
}

button{
  color: rebeccapurple;
  font-weight: bold;
  background-color: moccasin;
  border: 2px solid mediumseagreen;
}