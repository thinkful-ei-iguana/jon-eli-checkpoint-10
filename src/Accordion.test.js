import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Accordion from './Accordion.js';

configure({ adapter: new Adapter() });

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('<Accordion />', () => {
    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<Accordion sections={sections} />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });

    it('renders an empty li when the sections prop is not supplied', () => {
      const wrapper = shallow(<Accordion />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no sections as active by default', () => {
      const tree = renderer.create(<Accordion sections={sections} />).toJSON()
    expect(tree).toMatchSnapshot()
    })

    it(' opens a clicked section', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('button').at(1).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only opens the last section when multiple sections have been clicked.', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('button').at(0).simulate('click')
      wrapper.find('button').at(1).simulate('click')
      wrapper.find('button').at(2).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
  })
});