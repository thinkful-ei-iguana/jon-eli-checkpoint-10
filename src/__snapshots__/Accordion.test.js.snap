// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion />  opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders an empty li when the sections prop is not supplied 1`] = `
<ul>
  <li />
</ul>
`;

exports[`<Accordion /> renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders this UI as expected 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
